stl/inc/algorithm

```diff
@@ -2671,6 +2671,51 @@ namespace ranges {
 } // namespace ranges
 #endif // __cpp_lib_concepts
 
+#ifdef __cpp_lib_concepts
+namespace ranges {
+    // ALIAS TEMPLATE swap_ranges_result
+    template <class _In1, class _In2>
+    using swap_ranges_result = in_in_result<_In1, _In2>;
+
+    // VARIABLE ranges::swap_ranges
+    class _Swap_ranges_fn : private _Not_quite_object {
+    public:
+        using _Not_quite_object::_Not_quite_object;
+
+        // clang-format off
+        template <input_iterator _It1, sentinel_for<_It1> _Se1, input_iterator _It2, sentinel_for<_It2> _Se2>
+            requires indirectly_swappable<_It1, _It2>
+        constexpr swap_ranges_result<_It1, _It2>
+        operator()(_It1 _First1, _Se1 _Last1, _It2 _First2, _Se2 _Last2) const {
+            _Adl_verify_range(_First1, _Last1);
+            _Adl_verify_range(_First2, _Last2);
+            auto _UFirst1 = _Get_unwrapped(_STD move(_First1));
+            auto _ULast1 = _Get_unwrapped(_STD move(_Last1));
+            auto _UFirst2 = _Get_unwrapped(_STD move(_First2));
+            auto _ULast2 = _Get_unwrapped(_STD move(_Last2));
+            while ((_UFirst1 != _ULast1) && (_UFirst2 != _ULast2)) {

```

Comment:
Would a conventional for loop be viable here too?

Reply:
Yes, and `++_UFirst1, (void) ++_UFirst2` in a `for` loop would avoid the stylistic issue of having two statements on a single line below: `++_UFirst1; ++_UFirst2;`

(Our conventions currently permit multiple increments/decrements in `for` loops, as long as `(void)` casts are used to defend against overloaded comma operators with user-defined types, but in code elsewhere, clang-format puts separate statements on different lines. This makes it easier to notice when variables are being modified. The rationale is that `for` loops somewhat commonly need to increment/decrement multiple variables, so performing multiple modifications in a single line is reasonable there.)